pathway-analysis/qusage_meta_analysis.Rmd

```diff
@@ -0,0 +1,429 @@
+robinson_metadata_df <- readr::read_tsv(robinson_metadata_file) %>%
+  # drop columns that are all NA
+  dplyr::select(-which(apply(is.na(.), 2, all)))
+```
+
+To make this more comparable to the Northcott et al. dataset, we're going to
+remove the WNT subgroup and outliers such that we are comparing the SHH group
+to Group 3 and Group 4.
+
+```{r}
+robinson_metadata_df <- robinson_metadata_df %>%
+  # retain only pertinent columns
+  dplyr::select(refinebio_accession_code, refinebio_title, age, `m stage`,
+                subgroup) %>%
+  # make more comparable to Northcott dataset
+  dplyr::filter(!(subgroup %in% c("WNT", "SHH OUTLIER"))) %>%
+  # we'll use SHH vs. Other for our pathway analysis
+  dplyr::mutate(shh_v_other = dplyr::case_when(
+    subgroup != "SHH" ~ "Other",
+    TRUE ~ "SHH"
+  ))
+```
+
+Reorder expression data to match metadata.
+
+```{r}
+robinson_mat <- robinson_mat[, robinson_metadata_df$refinebio_accession_code]
+```
+
+## Pathway analysis
+
+### Read in KEGG pathways
+
+```{r}
+kegg_pathways <- read.gmt(kegg_file)
+```
+
+### Northcott et al.
+
+```{r}
+northcott_results <- qusage(eset = northcott_mat,
+                            labels = northcott_metadata_df$shh_v_other,
+                            contrast = "SHH-Other",
+                            geneSets = kegg_pathways)
+```
+
+Save the Northcott et al. results to file.
+
+```{r}
+northcott_results_file <- file.path("results", 
+                                    "Northcott_SHH-Other_QSarray.RDS")
+readr::write_rds(northcott_results, northcott_results_file)
+```
+
+### Robinson et al.
+
+```{r}
+robinson_results <- qusage(eset = robinson_mat,
+                           labels = robinson_metadata_df$shh_v_other,
+                           contrast = "SHH-Other",
+                           geneSets = kegg_pathways)
+```
+
+Save the Robinson et al. results to file.
+
+```{r}
+robinson_results_file <- file.path("results", 
+                                   "Robinson_SHH-Other_QSarray.RDS")
+readr::write_rds(robinson_results, robinson_results_file)
+```
+
+## Meta-analysis
+
+### Combine probability density function
+
+`combinePDFs` is a `qusage` function we can use for meta-analysis.
+This function accepts a list of `QSArray` results.
+
+```{r}
+results_list <- list(
+  Northcott = northcott_results,
+  Robinson = robinson_results
+)
+```
+
+Because there are more samples in the Northcott et al. dataset, it will be 
+weighted more highly when combining the distribution.
+
+```{r}
+combined_results <- combinePDFs(results_list)
+```
+
+Just as in a single dataset, we can extract relevant information from 
+the combined `QSArray` with built-in `qusage` functions.
+
+Let's look at the top 20 pathways with `qsTable`.
+
+```{r}
+qsTable(combined_results)
+```
+
+### Plotting
+
+We can plot the distributions with `plotCombinedPDF`.
+First let's plot KEGG Pathways in Cancer, which has elevated expression in
+the SHH group as it has a _positive_ fold change.
```

Comment:
Can you put a quick one sentence summary about what the x-axis of "Pathway Activity" represents?